src/dune_patch/dune_patch.ml

```diff
@@ -15,18 +15,15 @@ include struct
 end
 
 let re =
-  let line xs = Re.seq ((Re.bol :: xs) @ [ Re.eol ]) in
-  let followed_by_line xs = Re.seq [ Re.str "\n"; line xs ] in
-  Re.compile
-  @@ Re.seq
-       [ Re.alt
-           [ line [ Re.str {|--- a/|}; Re.group (Re.rep1 Re.notnl) ]
-           ; line [ Re.str {|--- |}; Re.group (Re.rep1 Re.notnl) ]
-           ]
-       ; Re.alt
-           [ followed_by_line [ Re.str {|+++ b/|}; Re.group (Re.rep1 Re.notnl) ]
-           ; followed_by_line [ Re.str {|+++ |}; Re.group (Re.rep1 Re.notnl) ]
-           ]
+  let open Re in
+  let line xs = seq ((bol :: xs) @ [ eol ]) in
+  let filename = group @@ rep1 @@ compl [ space ] in
+  let timestamp = rep notnl in
+  compile
+  @@ seq
+       [ line [ str {|--- |}; opt (str "a/"); filename; timestamp ]
+       ; str "\n"
+       ; line [ str {|+++ |}; opt (str "b/"); filename; timestamp ]
```

Comment:
This matches
```
--- a/foo
+++ foo
```
And I assume that the previous code would (rightly) reject this. Maybe worth keeping this behavior?

Reply:
The previous code would trip over this case pretty badly.
`+++ foo` would get interpreted as a dev/null, which means deleting the file `a/foo`.
Now it is understood as a name change, and `foo` will get updated with the contents of the patch